src/Fable.Transforms/Global/Babel.fs

```diff
@@ -219,32 +219,236 @@ module PrinterExtensions =
 /// If the node contains no information about the source location, the field is null;
 /// otherwise it is an object consisting of a start position (the position of the first character of the parsed source region)
 /// and an end position (the position of the first character after the parsed source region):
-type Node =
+type IPrintable =
     abstract Print: Printer -> unit
 
+type Node =
+    | Pattern of Pattern
+    | Program of Program
+    | Statement of Statement
+    | Directive of Directive
+    | ClassBody of ClassBody
+    | Expression of Expression
+    | SwitchCase of SwitchCase
+    | CatchClause of CatchClause
+    | ObjectMember of ObjectMember
+    | TypeParameter of TypeParameter
+    | TypeAnnotation of TypeAnnotation
+    | ExportSpecifier of ExportSpecifier
+    | ImportSpecifier of ImportSpecifier
+    | InterfaceExtends of InterfaceExtends
+    | ObjectTypeIndexer of ObjectTypeIndexer
+    | FunctionTypeParam of FunctionTypeParam
+    | ModuleDeclaration of ModuleDeclaration
+    | VariableDeclarator of VariableDeclarator
+    | TypeAnnotationInfo of TypeAnnotationInfo
+    | ObjectTypeProperty of ObjectTypeProperty
+    | ObjectTypeCallProperty of ObjectTypeCallProperty
+    | ObjectTypeInternalSlot of ObjectTypeInternalSlot
+    | TypeParameterDeclaration of TypeParameterDeclaration
+    | TypeParameterInstantiation of TypeParameterInstantiation
+
+    interface IPrintable with
+        member this.Print(printer) =
+            match this with
+            | Pattern(n) -> printer.Print(n)
```

Comment:
Now that we are doing this, maybe it's an opportunity to remove the `IPrintable` interface and just have `Print` methods in all the nodes, e.g.:

```fsharp
member this.Print(printer) =
    match this with
    | Pattern n -> n.Print(printer)
    ...
```

Maybe this also makes it easier to move the printer to another file to make the AST declarations cleaner.

Reply:
Yes at least most of them can be removed. There are still some problems related to e.g `PrintOptional`, `PrintCommaSeparatedArray` but I can see if I can fix it.

Reply:
I have now removed IPrintable. Next I will try to move the print handling into BabelPrinter.fs

Reply:
@alfonsogarciacaro But now is probably the right time to move Babel transformations i.e `IsProductiveStatement` back into Fable2Bable.fs? We don't want such transformations in the printer. I btw already did this for Babel2Python.fs